catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/ImportCommand.java

```diff
@@ -0,0 +1,256 @@
+import java.util.concurrent.atomic.AtomicReferenceFieldUpdater;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipInputStream;
+
+import org.apache.commons.io.FilenameUtils;
+import org.apache.karaf.shell.api.action.Argument;
+import org.apache.karaf.shell.api.action.Command;
+import org.apache.karaf.shell.api.action.Option;
+import org.apache.karaf.shell.api.action.lifecycle.Reference;
+import org.apache.karaf.shell.api.action.lifecycle.Service;
+import org.codice.ddf.catalog.transformer.zip.ZipValidator;
+import org.codice.ddf.commands.util.CatalogCommandRuntimeException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.io.ByteSource;
+
+import ddf.catalog.content.StorageProvider;
+import ddf.catalog.content.data.ContentItem;
+import ddf.catalog.content.data.impl.ContentItemImpl;
+import ddf.catalog.content.operation.impl.CreateStorageRequestImpl;
+import ddf.catalog.data.Metacard;
+import ddf.catalog.operation.impl.CreateRequestImpl;
+import ddf.catalog.transform.CatalogTransformerException;
+import ddf.catalog.transform.InputTransformer;
+import ddf.security.common.audit.SecurityLogger;
+
+@Service
+@Command(scope = CatalogCommands.NAMESPACE, name = "import", description = "Imports Metacards and history into the current Catalog")
+public class ImportCommand extends CatalogCommands {
+    private static final Logger LOGGER = LoggerFactory.getLogger(ImportCommand.class);
+
+    private static final int ID = 2;
+
+    private static final int TYPE = 3;
+
+    private static final int NAME = 4;
+
+    private static final int DERIVED_NAME = 5;
+
+    @Reference
+    private StorageProvider storageProvider;
+
+    @Argument(name = "Import File", description = "The file to import", index = 0, multiValued = false, required = true)
+    String importFile;
+
+    @Option(name = "--skip-signature-verification", required = false, multiValued = false, description =
+            "Exports the data but does NOT sign the resulting zip file. "
+                    + "WARNING: This file will not be able to be verified on import for integrity and authenticity.")
+    boolean unsafe = false;
+
+    @Override
+    protected Object executeWithSubject() throws Exception {
+        int metacards = 0;
+        int content = 0;
+        int derivedContent = 0;
+        ZipValidator zipValidator = initZipValidator();
+        File file = initImportFile(importFile);
+        InputTransformer transformer = getServiceByFilter(InputTransformer.class,
+                String.format("(%s=%s)",
+                        "id",
+                        DEFAULT_TRANSFORMER_ID)).orElseThrow(() -> new CatalogCommandRuntimeException(
+                "Could not get " + DEFAULT_TRANSFORMER_ID + " input transformer"));
+
+        if (unsafe) {
+            SecurityLogger.audit("Skipping validation check of imported data. There are no "
+                    + "guarantees of integrity or authenticity of the imported data");
```

Comment:
We should be better about this in general. The only way I can think of (off the top of my head anyways) to forge logs when using log4j would be to include CRLFs in the log message to make it look like a new log was written at the right time. Might be worth it to make sure we encode those correctly in the SecurityLogger so that folks don't need to worry about it. Might be worth a ticket, but I wouldn't try to fix it here.